pkg/controller/utils/clusterinfo.go

```diff
@@ -127,6 +127,50 @@ func GetClusterMasterInstances(kclient client.Client) ([]string, error) {
 	return ids, nil
 }
 
+// MasterInstance used to fill out TargetDescription
+// when calling registerTargetInput
+type MasterInstance struct {
+	AvailabilityZone string
+	IPaddress        string
+}
+
+// GetMasterInstancesAZsandIPs gets all the master instances AZs and IPs
+func GetMasterInstancesAZsandIPs(kclient client.Client) ([]MasterInstance, error) {
+	machineList := &machineapi.MachineList{}
+	listOptions := []client.ListOption{
+		client.InNamespace("openshift-machine-api"),
+		client.MatchingLabels{masterMachineLabel: "master"},
+	}
+	err := kclient.List(context.TODO(), machineList, listOptions...)
+	if err != nil {
+		return []MasterInstance{}, err
+	}
+
+	masterInstances := make([]MasterInstance, 0)
+	for _, mi := range machineList.Items {
+		// get the AZ from a Master object's providerSpec.
+		codec, err := awsproviderapi.NewCodec()
+		if err != nil {
+			return masterInstances, err
+		}
+		awsconfig := &awsproviderapi.AWSMachineProviderConfig{}
+		err = codec.DecodeProviderSpec(&machineList.Items[0].Spec.ProviderSpec, awsconfig)
+		if err != nil {
+			return masterInstances, err
+		}
+		az := awsconfig.Placement.AvailabilityZone
+
+		// get the IP address from Master object's status
+		ip := mi.Status.Addresses[0].Address
```

Comment:
It seems a little dangerous to me to just get the first element of the Addresses slice. Each machine will have 3 diff types of address (as example below). We're safer creating a quick helper function that will return the Address from the item that has `type == InternalIP`

Sample from Machine Status:
```yaml
status:
  addresses:
  - address: 10.0.138.191
    type: InternalIP
  - address: ip-10-0-138-191.ec2.internal
    type: InternalDNS
  - address: ip-10-0-138-191.ec2.internal
    type: Hostname
  lastUpdated: "2020-03-10T17:34:43Z"

```

Reply:
They could be in a diff order, hence my concern.

Reply:
removed func